src/EditorFeatures/Core/LanguageServer/AlwaysActiveLanguageClientEventListener.cs

```diff
@@ -60,8 +60,9 @@ private async Task LoadAsync()
                 using var token = _asynchronousOperationListener.BeginAsyncOperation(nameof(LoadAsync));
 
                 // Explicitly switch to the bg so that if this causes any expensive work (like mef loads) it 
-                // doesn't block the UI thread.
-                await TaskScheduler.Default;
+                // doesn't block the UI thread. Note, we always yield because sometimes our caller starts
+                // on the threadpool thread but is indirectly blocked on by the UI thread.
```

Comment:
We might want to put a comment here explaining this case a bit more since this is otherwise very not obvious without looking at this bug.